ros_ws/src/interpreter_pkg/interpreter_pkg/interpreter.py

```diff
@@ -0,0 +1,168 @@
+import os
+import rclpy
+from rclpy.node import Node
+from example_interfaces.msg import String
+
+import langchain
+from langchain.chains import LLMChain
+from langchain.chat_models import ChatOpenAI
+from langchain.llms import OpenAI
+from langchain.prompts import (ChatPromptTemplate, HumanMessagePromptTemplate,
+                                SystemMessagePromptTemplate)
+from langchain.schema import  HumanMessage, SystemMessage
+
+from ament_index_python.packages import get_package_share_directory
+
+data_dir = os.path.join(get_package_share_directory("interpreter_pkg"))
+OpenAI.api_key=os.environ["OPENAI_API_KEY"]
+MODEL_TO_USE="gpt-4"
+
+class LLM_Object:
+    def __init__(self, context_file: str, model:str, temperature:int):
+        with open(os.path.join(data_dir,context_file), 'r') as file:
+            context= file.read().replace('\n', '')
+        self.context=context
+        print("context is: "+self.context)
+        self.rolling_context=self.context
+        self.temp=temperature
+        self.chat=ChatOpenAI(model_name=model)
+        self.system_message=SystemMessage(content=self.context)
+
+    def reset_context(self):
+        self.rolling_context=[self.system_message]
+        
+class TerpNode(Node):
```

Comment:
"Terp" node? Weird name, which is fine-ish, but then it should have a comment explaining what this object is and what it's for

Reply:
I fixed it to InterpreterNode to be clear